### ISSUE TITLE
FR: display Variable definition from Working-Storage

### ISSUE THREAD

**Issue by eugeniodilo** [marcsosduma/cobgdb#26]
during a session of debugging and animating the procedure division code, it would be useful to be able to click on a variable to jump to the working-storage where the variable is defined to see its definition clearly and then be able to return with another click to the instruction in execution to continue with debugging.

for example:
- during the animation ... right click with the mouse on a variable
- a small pop-up window appears with two menu options:
    - displays the content of the variable  
    - jumps to the working-storage definition of the variable  
- click again with the right button: you return to the animation to continue with debugging ...

**Comment by GitMensch** [comment]
I agree and by coincidence I've created a FR to provide that in cobcd 21 hours ago :-)
https://gitlab.cobolworx.com/COBOLworx/cbl-gdb/-/issues/102

(you possibly find more relevant issues there, including closed ones, that you may want to test here)

The issue: there's likely no code that cobgdb can analyze for that - in this case it would have to use the symbol listing `-t- -Xref -fno-tsource`  (which commonly won't work as it doesn't provide the source names - and I have no clue why this never occurred to me).

Keep in mind that variables may be defined through `TYPEDEF`s and with `REPLACE` and most important: would commonly be in a copybook.

**Comment by eugeniodilo** [comment]
It might be simpler and better to add a new command (D) = show Definition, without touching the functioning of the right click on a variable which now executes the H command.

This new command D = show Definition, can be also activated with a click on the wheel (middle button) of the mouse.
If the user click on the middle button (the wheel) , the program open a  new window showing the working-storage definition of the field ... = the program will show the variable in the listing of the program .
After that the user can close the window and proceed with the debugging session.

**Comment by GitMensch** [comment]
An extra window like a separate process (in this case the user may configure an editor for that which would be called by a system call [if it is an X-Editor like gvim, then it would be in a separate window, otherwise in the same])?

And the harder part: What is your suggestion how to get to that definition in the first place?

**Comment by eugeniodilo** [comment]
For me it would work similarly to the H command.
Now H opens a window showing the contents of the variable.

The new command D would open a window in which it would show the listing which is an output of the compilation, positioning itself on the required field.
To position yourself on the required field perhaps you could use one of the GDB commands (does GDB have a "Find" command?)

**Comment by GitMensch** [comment]
> To position yourself on the required field perhaps you could use one of the GDB commands (does GDB have a "Find" command?)

GDB knows the definition and likely also the place of C variables, but doesn't know anything about COBOL.
As most variables will be found in a copybook it doesn't help to search from top of the current file either.

**Comment by eugeniodilo** [comment]
Ok I understand your comments well but searching from the beginning of the current file would still be a good solution even if not perfect and complete.
We have this program available which now has almost everything needed for debugging, we need to improve it as much as possible. Consequently, if it is something too complex then we can also not do it, it is not a serious problem.
During a debug session I can already scroll through the source on the screen and manually search for the field definition and then return to debugging.
However, since I can do this operation manually, I think it should be possible to find a way to automate it (without requiring absolute completeness or perfection, if there are limits and constraints just be aware of them).
